src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1325,6 +1325,23 @@ public synchronized void close() throws InterruptedException {
     }
 
     /**
+     * Close this client object as the {@link #close() } method.
+     * This method will wait for internal resources to be released.
+     *
+     * @param waitForShutdownTimeoutMs timeout (in milliseconds) to wait for resources to be released.
+     * Use zero or a negative value to skip the wait
+     * @throws InterruptedException
+     * @return true if waitForShutdownTimeout is greater than zero and all of the resources have been released
+     */
+    public boolean close(int waitForShutdownTimeoutMs) throws InterruptedException {
+        close();
+        if (waitForShutdownTimeoutMs > 0) {
+            return testableWaitForShutdown(waitForShutdownTimeoutMs);
```

Comment:
@hanm I did not change the testableWaitForShutdown method in order not to break binary compatibilty with existing subclasses.
For instance in my projects I use that method in order to achieve the purpose of this patches,
If it is not a problem for other users I will change the signature